src/routes.js

```diff
@@ -111,7 +111,7 @@ const transactionsHistory = (dbApi: DbApi, { logger, apiConfig }: ServerConfig)
   validateDatetimeReq(req.body)
   logger.debug('[transactionsHistory] request is valid')
   const result = await dbApi.transactionsHistoryForAddresses(
-    apiConfig.txHistoryResponseLimit,
+    req.body.txLimit || apiConfig.txHistoryResponseLimit,
```

Comment:
I would cap the "txLimit" by the txHistoryResponseLimit of the server as there are few outlying addresses that have a very long history and it may overload the server.

I know that an equivalent call we have in the "legacy-routes" sends back all the history of the address but this was to mimick the behavior of the cardanoexplorer and I would not introduce that behavior into new calls.

Reply:
Than an a `offset` should be the key, I guess, I will  what could be done here.

Reply:
also, please validate the `txLimit` field to ensure it's an integer

Reply:
also, don't forget to extend the `TxHistoryRequest` type to include the `txLimit` field. By the way, once you rebase to `new-importer`, you may leverage the "tx-ordinal" field: https://github.com/Emurgo/tangata-manu/blob/master/migrations/1567075489959_add-txs-ordinal.js

Reply:
bad news - I was checking the tx_ordinal field in our new-importer instance database and it does not seem to mean what I thought it does. It does not increase by one for each new transaction, it's mostly just zero or one as far as I checked so I don't know right now what it really means, but we can't use it in the way I proposed, sorry